rust_snuba/rust_arroyo/src/processing/strategies/produce.rs

```diff
@@ -0,0 +1,160 @@
+
+use log::warn;
+use crate::backends::Producer;
+use crate::backends::kafka::producer::KafkaProducer;
+use crate::backends::kafka::types::KafkaPayload;
+use crate::processing::strategies::{
+    CommitRequest, MessageRejected, ProcessingStrategy,
+};
+use crate::types::{Message, TopicOrPartition};
+use std::collections::VecDeque;
+use std::time::{Duration, Instant};
+
+pub struct Produce<TPayload: Clone + Send + Sync> {
+    pub producer: KafkaProducer,
+    pub next_step: Box<dyn ProcessingStrategy<TPayload>>,
+    queue: VecDeque<Message<TPayload>>,
+    topic: TopicOrPartition,
+    closed: bool,
+    max_queue_size: usize,
+}
+
+// TODO: make the queue store futures
+// TODO: make this generic
```

Comment:
Does it actually need to be generic? i.e. Any reason to take anything other than a KafkaPayload here?

Reply:
Ur right. removed